libs/blocks/global-navigation/utilities/keyboard/mainNav.js

```diff
@@ -0,0 +1,143 @@
+/* eslint-disable class-methods-use-this */
+import { selectors, getNextVisibleItem, getPreviousVisibleItem } from './utils.js';
+import Popup from './popup.js';
+import MobilePopup from './mobilePopup.js';
+import { closeAllDropdowns, expandTrigger } from '../utilities.js';
+
+class MainNavItem {
+  constructor() {
+    this.desktop = window.matchMedia('(min-width: 900px)');
+    this.popup = new Popup({ mainNav: this });
+    this.mobilePopup = new MobilePopup({ mainNav: this });
+    this.addEventListeners();
+  }
+
+  addEventListeners() {
+    document.querySelector('header').addEventListener('click', (e) => {
+      if (!e.target.closest(selectors.fedsNav) || e.target.closest(selectors.popup)) return;
+      const open = document.querySelector(selectors.expandedPopupTrigger);
+      closeAllDropdowns();
+      if (open !== e.target) this.open({ triggerEl: e.target });
+    });
+
+    document.querySelector('header').addEventListener('keydown', (e) => {
+      if (!e.target.closest(selectors.fedsNav) || e.target.closest(selectors.popup)) return;
+      this.setActive(e.target);
+      if (e.shiftKey && e.code === 'Tab') {
+        const open = document.querySelector(selectors.expandedPopupTrigger);
+        if (open) {
+          if (this.prev === -1) {
+            closeAllDropdowns();
+          } else {
+            e.preventDefault();
+            this.focusPrev({ focus: 'last' });
+          }
+        }
+        return;
+      }
+
+      switch (e.code) {
+        case 'Escape': {
+          closeAllDropdowns();
+          break;
+        }
+        // TODO popup navigation logic.
+        case 'ArrowLeft': {
+          if (document.dir === 'ltr') {
+            if (this.prev === -1) break;
+            this.focusPrev({ focus: null });
+          } else {
+            if (this.next === -1) break;
+            this.focusNext({ focus: null });
+          }
+          break;
+        }
+        case 'ArrowUp': {
```

Comment:
I'm seeing some weird behavior with up and down arrows, here's how to reproduce:
1. go to your after page
2. scroll down
3. focus on the last element in the nav - "no dropdown"
4. press "up" arrow 3-4 times
5. press "down" arrow 3-4 times

desired behavior: no page scrolling
actual behavior: focus is changed to previous / next elements, but pressing the arrows will introduce page scrolling as well

Reply:
Maybe `stopPropagation` was missed here, I saw it in the popups logic.

Reply:
In the AEM implementation, only for the main nav elements the event stops doing anything - e.g. pressing down on the search will still scroll.

I'll add it that way, I think that makes the most sense, since those are the only elements that interact with the arrow keys